llama-index-integrations/llms/llama-index-llms-vllm/llama_index/llms/vllm/base.py

```diff
@@ -375,23 +375,25 @@ def stream_complete(
         kwargs = kwargs if kwargs else {}
         params = {**self._model_kwargs, **kwargs}
 
-        from vllm import SamplingParams
-
-        # build sampling parameters
-        sampling_params = SamplingParams(**params).__dict__
+        sampling_params = dict(**params)
         sampling_params["prompt"] = prompt
         response = post_http_request(self.api_url, sampling_params, stream=True)
 
         def gen() -> CompletionResponseGen:
             response_str = ""
+            prev_prefix_len = len(prompt)
             for chunk in response.iter_lines(
                 chunk_size=8192, decode_unicode=False, delimiter=b"\0"
             ):
                 if chunk:
                     data = json.loads(chunk.decode("utf-8"))
 
-                    response_str += data["text"][0]
-                    yield CompletionResponse(text=response_str, delta=data["text"][0])
+                    increasing_concat = data["text"][0]
```

Comment:
this API sends increasing output no need to repeat it again. see https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/api_server.py#L56

I have an assert in my test checking that prompt occur only once